spirv_msl.cpp

```diff
@@ -393,6 +428,15 @@ uint32_t CompilerMSL::add_interface_block(StorageClass storage)
 	MemberSorter memberSorter(ib_type, meta[ib_type_id], sort_aspect);
 	memberSorter.sort();
 
+	// Sort input or output variables alphabetical
+	auto &execution = get_entry_point();
+	if ((execution.model == ExecutionModelFragment && storage == StorageClassInput) ||
+	    (execution.model == ExecutionModelVertex && storage == StorageClassOutput))
+	{
+		MemberSorter memberSorter(ib_type, meta[ib_type.self], MemberSorter::Alphabetical);
+		memberSorter.sort();
+	}
```

Comment:
While there is nothing wrong with sorting alphabetically, Metal won't use that to match vertex output and fragment input (it matches names regardless of order (maybe you've encountered an MSL bug?)...and the names may be different if coming from different sources). `CompilerMSL` uses the `[[user(locnN)]]` qualifier to ensure matching. See (2) above...SPIR-V should be providing this location value (or you could set it earlier in the processing).